### PR TITLE
Feature/main-page-data-and-other-changes

### PR DESCRIPTION
**Pull request by antyas**

### CODE REVIEW

File: src/lib/time.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export const elapsed = {
+  allMinutes: getAllElapsedMinutes,
+  seconds: getElapsedSeconds,
+};
+
+function formatXX(item: number) {
+  return item < 10 ? '0' + item : item;
+}
+
+export function format(timestamp: string) {
```
Comment (0x009922) [inline src/lib/time.ts:24]: I'd recommend to define an [opaque](https://github.com/sindresorhus/type-fest/blob/main/source/opaque.d.ts) `Timestamp` type to make types stronger:

```ts
type Timestamp = Opaque<string, 'ISO timestamp'>
```

File: src/views/AssetsList.vue [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  <BaseContentBlock :title="$t('assets')" class="assets-list-page">
+    <BaseTable
+      :loading="table.loading.value"
+      :pagination="table.pagination"
+      :items="table.items.value"
+      @next-page="table.nextPage()"
+      @prev-page="table.prevPage()"
+      @set-page="table.setPage($event)"
+      @set-size="table.setSize($event)"
+    >
```
Comment (0x009922) [inline src/views/AssetsList.vue:11]: You can refactor `BaseTable` to accept just what `useTable()` returns so you don't need to configure all these props:

```html
<BaseTable :table="table" />
```

File: src/views/BlocksList.vue [hunk 3]
```diff
@@ -15,7 +15,60 @@
...
-          <span class="h-sm cell">{{ $t('validators') }}</span>
+        </div>
+      </template>
+
+      <template #row="{ item }: { item: Block }">
+        <div class="blocks-list-page__row">
+          <span class="cell">
+            <a :href="`/blocks/${item.height}`" class="primary-link">
+              {{ item.height }}
+            </a>
```
Comment (0x009922) [inline src/views/BlocksList.vue:26]: Wait, why not `<RouterLink>`? It's not an SPA pattern to reload page on navigation.

File: src/views/BlocksList.vue [hunk 4]
```diff
@@ -15,7 +15,60 @@
...
+
+          <div class="cell">
+            <time class="row-text">{{ format(item.timestamp) }}</time>
+          </div>
+
+          <span class="cell">
+            <BaseCopyRow :name="$t('hash')" :value="item.block_hash">
+              <a :href="`/blocks/${item.height}`" class="primary-link">
+                {{ item.block_hash }}
+              </a>
```
Comment (0x009922) [inline src/views/BlocksList.vue:37]: Expect `<RouterLink>` here as well

File: src/views/BlocksList.vue [hunk 5]
```diff
@@ -15,7 +15,60 @@
...
+        </div>
+      </template>
+
+      <template #mobile-card="{ item }: { item: Block }">
+        <div class="blocks-list-page__mobile-card">
+          <div class="blocks-list-page__mobile-row">
+            <span class="h-sm blocks-list-page__mobile-label">{{ $t('height') }}</span>
+            <a :href="`/blocks/${item.height}`" class="primary-link">
+              {{ item.height }}
+            </a>
```
Comment (0x009922) [inline src/views/BlocksList.vue:51]: Expect `<RouterLink>` here as well

File: src/views/BlocksList.vue [hunk 6]
```diff
@@ -15,7 +15,60 @@
...
+            <span class="h-sm blocks-list-page__mobile-label">{{ $t('age') }}</span>
+            <time class="row-text">{{ format(item.timestamp) }}</time>
+          </div>
+
+          <div class="blocks-list-page__mobile-row">
+            <span class="h-sm blocks-list-page__mobile-label">{{ $t('hash') }}</span>
+            <BaseCopyRow :name="$t('hash')" :value="item.block_hash">
+              <a :href="`/blocks/${item.height}`" class="primary-link">
+                <ShortHash :hash="item.block_hash" />
+              </a>
```
Comment (0x009922) [inline src/views/BlocksList.vue:64]: Expect `<RouterLink>` here as well

File: src/views/DomainsList.vue [hunk 7]
```diff
@@ -11,9 +11,52 @@
...
+          <span class="h-sm cell">{{ $t('accounts') }}</span>
+        </div>
+      </template>
+
+      <template #row="{ item }: { item: Domain }">
+        <div class="domains-list-page__row">
+          <span class="cell">
+            <a :href="`/domains/${item.id}`" class="primary-link">
+              {{ item.id }}
+            </a>
```
Comment (0x009922) [inline src/views/DomainsList.vue:26]: Expect `<RouterLink>` here as well

File: src/views/DomainsList.vue [hunk 8]
```diff
@@ -11,9 +11,52 @@
...
+        </div>
+      </template>
+
+      <template #mobile-card="{ item }: { item: Domain }">
+        <div class="domains-list-page__mobile-card">
+          <div class="domains-list-page__mobile-row">
+            <span class="h-sm domains-list-page__mobile-label">{{ $t('name') }}</span>
+            <a :href="`/domains/${item.id}`" class="primary-link">
+              {{ item.id }}
+            </a>
```
Comment (0x009922) [inline src/views/DomainsList.vue:43]: Expect `<RouterLink>` here as well

**Review by 0x009922 — CHANGES_REQUESTED** [review]
LGTM, expect the moment of using raw `<a>` instead of `<RouterLink>`

File: src/views/AssetsList.vue (hunk 2 above)
Comment (antyas) [reply src/views/AssetsList.vue:11]: I left the possibility for custom logic for tables. but probably in this project such freedom is not required. I will redo it in the next pr

File: src/lib/time.ts (hunk 1 above)
Comment (antyas) [reply src/lib/time.ts:24]: I do not think that this will add reliability, but I probably will use it later

File: src/views/BlocksList.vue (hunk 3 above)
Comment (antyas) [reply src/views/BlocksList.vue:26]: I will definitely fix this in future pr. I'm shocked how could I let this happen.

**Review by 0x009922 — APPROVED** [review]

**Pull request merged**